src/general.py

```diff
@@ -193,10 +196,19 @@ def execute_tests(tests_dir: Path, working_dir: Path, test_list: list):
 
         os.chdir(execute_dir)
         job = generate_sub_object(sub_file, test.name, abs_timestamp_dir)
-        item_data = [
-            {"ResourceName": resource, "uniq_output_dir": f"results/{resource}"}
-            for resource in resources.keys()
-        ]
+
+        item_data = []
+        for resource in resources.keys():
+            resource_size = resources[resource]
+            sample_size = ceil(resource_size * sample_percent)
+            for i in range(sample_size):
+                item = {
+                    "ResourceName": resource,
+                    "resource_dir": f"results/{resource}",
```

Comment:
We should make sure to update the documentation to state the macro `$(resource_dir)` can be used for output on a per resource basis rather then the original `$(uniq_output_dir)`